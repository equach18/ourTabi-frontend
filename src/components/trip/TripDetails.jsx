function TripDetails({ trip }) {
  return (
    <div className="space-y-2 text-gray-700">
      <p>
        <strong>Destination:</strong> {trip.destination}
      </p>
      {trip.startDate && (
        <p>
          <strong>Dates:</strong>{" "}
          {new Date(trip.startDate).toLocaleDateString()} -{" "}
          {new Date(trip.endDate).toLocaleDateString()}
        </p>
      )}
      <p>
        <strong>Privacy:</strong> {trip.isPrivate ? "Private" : "Public"}
      </p>
    </div>
  );
}

export default TripDetails;
