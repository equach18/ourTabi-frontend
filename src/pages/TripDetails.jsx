import { useContext, useEffect } from "react";
import TripContext from "../context/TripContext";
import { Link, useNavigate } from "react-router-dom";
import MemberPanel from "../components/member/MemberPanel";
import CommentPanel from "../components/comment/CommentPanel";
import ActivityList from "../components/activity/ActivityList";

function TripDetails({ currentUser, removeTrip }) {
  const { trip, activities, members } = useContext(TripContext);
  const navigate = useNavigate();

  useEffect(() => {
    if (!trip) {
      navigate("/not-found");
    }
  }, [trip, navigate]);

  // handle limited access on trip details if user is not  a member
  const isMember = members.some((m) => m.userId === currentUser.id);
  // handle limited access if user is not the owner
  const isTripCreator = trip.creatorId === currentUser.id;

  async function handleDelete() {
    if (window.confirm("Are you sure you want to delete this trip?")) {
      await removeTrip(trip.id);
      navigate("/dashboard");
    }
  }

  return (
    <div className="container mx-auto mt-10 p-6">
      {/* Trip Header */}
      <div className="flex justify-between items-center">
        <h1 className="text-3xl font-bold">{trip.title}</h1>
        {isTripCreator && (
          <div className="space-x-4">
            <Link
              to={`/trips/${trip.id}/edit`}
              className="bg-blue-500 text-white px-4 py-2 rounded-lg"
            >
              Edit Trip
            </Link>
            <button
              onClick={handleDelete}
              className="bg-red-500 text-white px-4 py-2 rounded-lg"
            >
              Delete Trip
            </button>
          </div>
        )}
      </div>

      {/* Trip Details */}
      <p className="text-gray-600 mt-4">
        <strong>Destination:</strong> {trip.destination}
      </p>
      {trip.startDate && (
        <p className="text-gray-600">
          <strong>Dates:</strong>{" "}
          {new Date(trip.startDate).toLocaleDateString()} -{" "}
          {new Date(trip.endDate).toLocaleDateString()}
        </p>
      )}
      <p className="text-gray-600">
        <strong>Privacy:</strong> {trip.isPrivate ? "Private" : "Public"}
      </p>

      {/* Activities Section */}
      <div className="mt-8">
        <ActivityList activities={activities} trip={trip} />
      </div>

      {/* Members Section */}
      {isMember && (
        <div className="mt-8">
          <MemberPanel isTripCreator={isTripCreator} members={members} />
        </div>
      )}

      {/* Comments Section */}
      {isMember && (
        <div className="mt-8">
          <h2 className="text-2xl font-semibold">Comments</h2>
          <CommentPanel />
        </div>
      )}
    </div>
  );
}

export default TripDetails;
