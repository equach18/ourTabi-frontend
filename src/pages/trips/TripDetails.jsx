import { useContext, useEffect } from "react";
import TripContext from "../../context/TripContext";
import { Link, useNavigate } from "react-router-dom";
// import MembersList from "./MembersList";
// import CommentsSection from "./CommentsSection";
import ActivityList from "./ActivityList";

function TripDetails({ currentUser, removeTrip }) {
  
  const { trip, activities } = useContext(TripContext);
  const navigate = useNavigate();

  useEffect(() => {
    if (!trip) {
      navigate("/not-found");
    }
  }, [trip, navigate]);

  // handle limited access on trip details if user is not a a member
  const isTripCreator = trip.creatorId === currentUser.id;

  async function handleDelete() {
    if (window.confirm("Are you sure you want to delete this trip?")) {
      await removeTrip(trip.id);
      navigate("/dashboard");
    }
  }

  return (
    <div className="container mx-auto mt-10 p-6">
      {/* Trip Header */}
      <div className="flex justify-between items-center">
        <h1 className="text-3xl font-bold">{trip.title}</h1>
        {isTripCreator && (
          <div className="space-x-4">
            <Link
              to={`/trips/${trip.id}/edit`}
              className="bg-blue-500 text-white px-4 py-2 rounded-lg"
            >
              Edit Trip
            </Link>
            <button
              onClick={handleDelete}
              className="bg-red-500 text-white px-4 py-2 rounded-lg"
            >
              Delete Trip
            </button>
          </div>
        )}
      </div>

      {/* Trip Details */}
      <p className="text-gray-600 mt-4">
        <strong>Destination:</strong> {trip.destination}
      </p>
      {trip.startDate && (
        <p className="text-gray-600">
          <strong>Dates:</strong>{" "}
          {new Date(trip.startDate).toLocaleDateString()} -{" "}
          {new Date(trip.endDate).toLocaleDateString()}
        </p>
      )}
      <p className="text-gray-600">
        <strong>Privacy:</strong> {trip.isPrivate ? "Private" : "Public"}
      </p>

      {/* Members Section */}
      {/* <MembersList /> */}

      {/* Activities Section */}
      <div className="mt-8">
        <ActivityList activities={activities} trip={trip} />
      </div>

      {/* Comments Section */}
      {/* <div className="mt-8">
        <h2 className="text-2xl font-semibold">Comments</h2>
        <CommentsSection />
      </div> */}
    </div>
  );
}

export default TripDetails;
