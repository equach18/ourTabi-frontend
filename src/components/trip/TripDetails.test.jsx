import { render, screen } from "@testing-library/react";
import TripDetails from "./TripDetails";

describe("TripDetails", () => {
  const testTrip = {
    destination: "Chicago",
    isPrivate: true,
  };

  it("matches snapshot", () => {
    const { asFragment } = render(<TripDetails trip={testTrip} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it("renders destination and privacy", () => {
    render(<TripDetails trip={testTrip} />);
    expect(screen.getByText(/destination:/i)).toBeInTheDocument();
    expect(screen.getByText(/chicago/i)).toBeInTheDocument();
    expect(screen.getByText(/privacy:/i)).toBeInTheDocument();
    expect(screen.getByText(/private/i)).toBeInTheDocument();
  });

  it("does not render dates if startDate is missing", () => {
    render(<TripDetails trip={testTrip} />);
    expect(screen.queryByText(/dates:/i)).not.toBeInTheDocument();
  });
});
