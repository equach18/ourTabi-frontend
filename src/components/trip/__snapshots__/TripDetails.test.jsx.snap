// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`TripDetails > matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="space-y-2 text-gray-700"
  >
    <p>
      <strong>
        Destination:
      </strong>
       Chicago
    </p>
    <p>
      <strong>
        Privacy:
      </strong>
       Private
    </p>
  </div>
</DocumentFragment>
`;
